ldms/src/core/ldms_xprt.c

```diff
@@ -3764,20 +3764,27 @@ int ldms_xprt_sockaddr(ldms_t _x, struct sockaddr *local_sa,
 {
 	struct ldms_xprt *x = _x;
 	zap_err_t zerr;
- again:
-	if (x->sa_len) {
-		/* use cached addresses */
-		*sa_len = x->sa_len;
-		memcpy(local_sa, &x->local_sa, x->sa_len);
-		memcpy(remote_sa, &x->remote_sa, x->sa_len);
-		return 0;
-	}
-	zerr = zap_get_name(x->zap_ep, &x->local_sa, &x->remote_sa, &x->sa_len);
-	if (zerr) {
-		x->sa_len = 0;
-		return -1;
+	socklen_t __sa_len = *sa_len;
+
+	if (*sa_len == 0)
+		return EINVAL;
+
+	if (!x->sa_len) {
```

Comment:
Why are you checking this? I'm actually not sure why sa_len is in the transport structure at all. The input value is always going to be sizeof(local/remote_sa). 

The local_sa and remote_sa addresses should be sockaddr_storage (128B), not sockaddr(14B). The current definitions are too small to accept an IPv6 address. Right now it's not symptomatic because we're not using IPv6.
That said, it's not the correct way to handle addresses like this.